### PR TITLE
Add static_graph arg to DistributedDataParallelKwargs.

### PR DESCRIPTION
**Pull request by rom1504**
supported by https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html

This is particularly useful when using gradient checkpointing

See https://discuss.pytorch.org/t/ddp-and-gradient-checkpointing/132244/3 for more details

### CODE REVIEW

**Comment by muellerzr** [comment]
Thanks! Could we add a warning to this please similar to the other one that exists for these kwargs mentioning its only available on pytorch>=1.11?

**Comment by rom1504** [comment]
Sure, done

**Comment by muellerzr** [comment]
@rom1504 looks like the last thing we need is to run `make style; make quality` :)

**Comment by rom1504** [comment]
@muellerzr fixed

**Comment by muellerzr** [comment]
Great! Thanks!!
